proto/directions.proto

```diff
@@ -7,10 +7,18 @@ import public "sign.proto";
 message DirectionsLeg {
 
   message Summary {
-    optional float length = 1;          // kilometers or miles based on units
-    optional double time = 2;           // seconds
-    optional BoundingBox bbox = 3;      // Bounding box of the shape
-    optional bool has_time_restrictions = 4; // Does the route contain any time restrictions?
+    oneof has_length {
+      float length = 1;          // kilometers or miles based on units
+    }
```

Comment:
this is the main change, every scalar type that was `optional` which was all of them, needs to become a `oneof`